java_src/foam/nanos/logger/NanoLogger.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.logger;
+
+import foam.core.*;
+import foam.nanos.NanoService;
+import java.util.logging.*;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.io.IOException;
+
+public class NanoLogger extends ContextAwareSupport implements NanoService {
+  private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy.MM.dd.HH.mm.ss");
```

Comment:
SimpleDateFormat is not thread-safe. So, if the logger is reentrant, this will also have to change. We should make a ThreadSafeDateFormat using ThreadLocal SimpleDateFormat objects.